testing/extended/verify_env.py

```
@@ -0,0 +1,290 @@
+        port_int = int(port)
+    except ValueError:
+        return f"MAIL_SMTP_PORT is not numeric: {port!r}"
+    try:
+        with smtplib.SMTP(host, port_int, timeout=10) as s:
+            s.ehlo()
+            s.starttls()
+            s.login(user, password)
+            # Don't send anything — connecting + auth is enough.
+    except smtplib.SMTPException as e:
+        return f"SMTP login/connect failed: {e!s}"
+    except (OSError, socket.timeout) as e:
```

Ruff (0.15.13)

[warning] Replace aliased errors with `TimeoutError`

Replace with builtin `TimeoutError`

(UP041)